handsontable/src/3rdparty/walkontable/src/utils/stylesHandler.js

```diff
@@ -0,0 +1,182 @@
+/**
+ * Handles the theme-related style operations.
+ */
+export class StylesHandler {
+  /**
+   * The instance's root element.
+   *
+   * @type {HTMLElement}
+   */
+  #rootElement;
+
+  /**
+   * The computed style of the root element.
+   *
+   * @type {CSSStyleDeclaration}
+   * @private
+   */
+  #rootComputedStyle;
+
+  /**
+   * The root document of the instance.
+   *
+   * @type {Document}
+   * @private
+   */
+  #rootDocument;
+
+  /**
+   * `true` if the classic theme is enabled, `false` otherwise.
+   *
+   * @type {boolean}
+   */
+  #isClassicTheme = false;
+
+  /**
+   * An object to store CSS variable values.
+   *
+   * @type {object}
+   * @private
+   */
+  #cssVars = {};
+
+  /**
+   * Stores the computed styles for various elements.
+   *
+   * @type {object} - An object containing the computed styles if a nested structore of `element: { [element type]: {property: value} }`.
+   * @private
+   */
+  #computedStyles = {};
+
+  /**
+   * Initializes a new instance of the `StylesHandler` class.
+   *
+   * @param {object} domBindings - The DOM bindings for the instance.
+   */
+  constructor(domBindings) {
+    this.#rootElement = domBindings.rootTable.parentElement;
+    this.#rootComputedStyle = getComputedStyle(this.#rootElement);
+    this.#rootDocument = domBindings.rootDocument;
+
+    if (this.#rootComputedStyle?.getPropertyValue('--ht-line-height') === '') {
+      this.#isClassicTheme = true;
+    }
+
+    // Some of the properties are not stored in the CSS variables, so we need to get them from the computed style.
+    this.#getStylesForTD((computedStyle) => {
+      this.#computedStyles = {
+        ...this.#computedStyles,
+        ...{
+          td: {
+            'box-sizing': computedStyle['box-sizing'],
+            'border-top-width': parseInt(computedStyle['border-top-width'], 10),
+            'border-bottom-width': parseInt(computedStyle['border-bottom-width'], 10),
+          },
+        },
+      };
+    });
+  }
```

Comment:
I did it that way, because the computed styles object that would be returned from the function is modified after the function removes the temporary elements: https://jsfiddle.net/js_ziggle/m1kz078j/

I can rewrite the method so it returns just the css values requested as the function argument 👍